var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -91,9 +91,18 @@ def setup_build_environment(self, env):
         ar = which("ar", required=True)
         env.set("AR", ar.path)
 
+        output = Executable("openssl")("version", "-d", output=str, error=str)
+        ossl = self.spec["openssl"]
+        m = re.match('OPENSSLDIR: "([^"]+)"', output)
+        if m:
+            cert_dir = m.group(1)
+        elif ossl.external and ossl.prefix == "/usr":
+            cert_dir = "/etc"
+        else:
+            cert_dir = join_path(ossl.prefix, "etc")
         # Manually inject the path of openssl's certs for build.
-        certs = join_path(self.spec["openssl"].prefix, "etc/openssl/cert.pem")
-        env.set("CARGO_HTTP_CAINFO", certs)
+        certs = join_path(cert_dir, "openssl/cert.pem")
```

Comment:
Looks like this line is causing flake8 to grumble since certs isn't used. I think to fix we'll just need to comment while you test.

Reply:
Thanks, you're probably right.  Really want to figure this out, rust seems like the last holdout, and I thought it was going fine when we had it solved in LC.  No clue why it's failing in gitlab though.

Reply:
Well @alecbcs, good news is it's building, bad news is it only seems to build if we remove the env var setting. https://gitlab.spack.io/spack/spack/-/jobs/9050780

My current theory, based on the error message, is that there may be an assumption of a maximum path length in cargo for that var, because it only prints part of it.  It may be getting sliced and then causing it not to consider the defaults because the var was set?  I'm debating how to deal with this, we could have a rust variant, or some way to decide we don't want to set the var in the package maybe?